nautobot/core/ui/object_detail.py

```diff
@@ -483,17 +485,26 @@ def get_extra_context(self, context):
         for listing and adding objects. It also handles field inclusion/exclusion and
         displays the appropriate table title if provided.
         """
-        body_content_table = context[self.table_key]
+        if isinstance(context[self.table_key], tuple):
+            body_content_table_class, body_content_table_queryset = context[self.table_key]
+            body_content_table = body_content_table_class(body_content_table_queryset)
+        else:
+            body_content_table = context[self.table_key]
         request = context["request"]
 
         if self.exclude_fields or self.include_fields:
-            for column in body_content_table.columns:
-                if (self.exclude_fields and column.name in self.exclude_fields) or (
-                    self.include_fields and column.name not in self.include_fields
+            for column in body_content_table._sequence:
+                if (self.exclude_fields and column in self.exclude_fields) or (
+                    self.include_fields and column not in self.include_fields
                 ):
-                    body_content_table.columns.hide(column.name)
+                    body_content_table.columns.hide(column)
                 else:
-                    body_content_table.columns.show(column.name)
+                    body_content_table.columns.show(column)
+        else:
+            # If no exclude and include fields are specified
+            # show all columns
```

Comment:
Let me know if this makes sense

Reply:
Is this not the automatic/default behavior?

Reply:
(bump)